#150749930 Paystack Integration

noordean opened the pull request:
#### What does this PR do?
It integrates paystack payment system
#### Description of Task to be completed?
Integrate paystack payment system
#### How should this be manually tested?
By running `npm run test-local`
#### Any background context you want to provide?
Ensure selenium server and chromedriver are installed and make sure admin details are set in the .env file.
#### What are the relevant pivotal tracker stories?
#150749930
#### Screenshots (if appropriate)
<img width="1274" alt="screen shot 2017-09-21 at 8 15 40 am" src="https://user-images.githubusercontent.com/14357823/30694921-05b3d606-9ece-11e7-9936-d39e247f2918.png">

<img width="1266" alt="screen shot 2017-09-21 at 8 16 12 am" src="https://user-images.githubusercontent.com/14357823/30694930-0d1daa84-9ece-11e7-86ac-d53000049e06.png">

<img width="1172" alt="screen shot 2017-09-21 at 8 18 06 am" src="https://user-images.githubusercontent.com/14357823/30694933-14164cd8-9ece-11e7-807c-183cfa17cfb3.png">

<img width="1259" alt="screen shot 2017-09-21 at 8 16 31 am" src="https://user-images.githubusercontent.com/14357823/30694943-1d91524e-9ece-11e7-9a2a-fdec1c5817ae.png">


#### Questions:
Nil

noordean committed:
feature(integration): integrate paystack

- create a paystackPaymentForm view
- download minified paystackApi.js file from paystack.co
- save both the secretKey and publicKey in reaction.json
- set secretKey as Authorization in get request headers
- consume /transaction/verify/ endpoint, and supply referenceId as param

[Delivers]

noordean committed:
Merge branch 'staging' of https://github.com/andela/fengshui-rc into feature/#150749930/integrate-paystack
update integrate-paystack with staging

noordean committed:
bug(bug-fix): fix console bug

- fix shopId not found

[Fixes]

noordean committed:
Merge branch 'feature/#150749930/integrate-paystack' of https://github.com/andela/fengshui-rc into feat/#150749930/integrate-paystack
update integrate-paystack with remote

noordean committed:
bug(bug-fix): fix paystack secret keys

- add pauldowman:dotenv to meteor
- write paystack payment test

[Fixes]

noordean committed:
resolve merge conflict

temilaj commented on tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js:
```diff
@@ -0,0 +1,51 @@
+  const browserConfig = yaml.safeLoad(fs.readFileSync("./tests/acceptance-tests/config/settings.yml", "utf8"));
+  const baseUrl = browserConfig.base_url.toString();
+  browser.url(baseUrl);
+  // browser.getSession().then(function (sessionid) {
```
why do you have commented-out code in source control?

noordean replied:
I will fix that right now.

temilaj commented on tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js:
```diff
@@ -0,0 +1,51 @@
+    browser.click("#pay-btn");
+    browser.waitForExist(".order-item");
+    expect(browser.getAttribute("div", "order-item")).to.exist;
+  });
```
Is this test complete?. How do we know that the payment will be successful

noordean replied:
If the payment is successful, a div element with class "order-item" gets rendered, it shows a list of ordered products. So without the payment being successful, there is no way the element would exist

temilaj replied:
Do you think it'll be valuable to check what's inside the div?

noordean replied:
It's not checking for what is inside the div. It's checking if a div that has 'order-item' in one of its attributes exists. So it's checking for the existence of the div

temilaj commented on imports/plugins/included/paystack/lib/api/paystack.js:
```diff
@@ -0,0 +1,24 @@
+
+export const Paystack = {};
+
+const paystackHeaders = (secret) => {
```
Consider improving the documentation in your PR. Almost all funcitons do not have any form of documentation

noordean replied:
I will fix this right now

noordean committed:
bug(bug-fix): fix bugs on e2e test

                - mock the paystack iframe
                - add id's to complete-button and item element
                - add JSdoc comments
                - rectify the commented line e2e test

                [Fixes]
